index.js

```diff
@@ -1,32 +1,110 @@
+    builder.hook('before scripts', function(pkg, next) {
+        var options = {
+            bare: true
+        };
 
-    // No scripts field in the component.json file
-    if (pkg.config.scripts === undefined) return next();
+        if (pkg.dev) {
+            //For some reason dev-settings are not applied to all files.
+            devFlag = true;
+        }
+        if (devFlag === true) {
+            pkg.dev = devFlag;
+            options.sourceMap = true;
+        }
 
-    // Get all the coffee files from the scripts list
-    var coffee = pkg.config.scripts.filter(function(file){
-      return path.extname(file) === '.coffee';
-    });
+        // No scripts field in the component.json file
+        if (pkg.config.scripts === undefined) return next();
 
-    // No scripts
-    if( coffee.length === 0 ) return next();
-
-    coffee.forEach(function(file, i){
-      var realpath = pkg.path(file);
-      var str = fs.readFileSync(realpath, 'utf8');
-      var compiled = coffeescript.compile(str, { filename : realpath, bare: true });
-      pkg.removeFile('scripts', file);
-      pkg.addFile('scripts', file, compiled);
-      // This duplicates the code in the built package, but needed to be able to require
-      // modules without adding .coffee - a better solution is needed for production:
-      pkg.addFile('scripts', file.replace('.coffee', '.js'), compiled);
-    });
+        // Get all the coffee files from the scripts list
+        var coffee = pkg.config.scripts.filter(function(file) {
+            return path.extname(file) === '.coffee';
+        });
 
-    next();
-  });
-};
+        // No scripts
+        if (coffee.length === 0) return next();
+
+        coffee.forEach(function(file, i) {
+            options.filename = pkg.path(file);
+
+            var str = fs.readFileSync(options.filename, 'utf8');
+            var hash = calculate(str);
+            var hashFile = cacheDirPath + options.filename + '#' + hash + '.cache';
+            var buildFile = cacheDirPath + options.filename + '.js.cache';
+            var compiled = '';
+
+            try {
+                //See if the file has been compiled already
+                // If yes, use the compiled content
+                fs.openSync(hashFile, 'r');
+                debug(hashFile, ' from cache');
+                compiled = fs.readFileSync(buildFile, 'utf8');
+            } catch (err) {
+                //Else compile new content, create compiled cache-file and hash-file.
+                console.log('Compiling: ', options.filename);    
+                compiled = coffeescript.compile(str, options);
+                if(compiled.v3SourceMap){
+                    //Support for sourcemaps, whenever the doublicate content ticket is resolved
+                    compiled =  compiled.js; // + "\n//# sourceMappingURL=data:application/json;base64," + (btoa(unescape(encodeURIComponent(compiled.v3SourceMap)))) + "\n//# sourceURL=coffeescript";
```

Comment:
Whitespace (=  compiled.js), and what is the commented out stuff about btoa etc..?